src/Files.Uwp/Filesystem/BaseStorage/IBaseStorageFolder.cs

```diff
@@ -38,4 +39,11 @@ public interface IBaseStorageFolder : IStorageItem2, IStorageFolder, IStorageFol
         new BaseStorageFileQueryResult CreateFileQueryWithOptions(QueryOptions queryOptions);
         new BaseStorageFolderQueryResult CreateFolderQueryWithOptions(QueryOptions queryOptions);
     }
+
+    public interface ICreateFileWithStream
```

Comment:
Is it needed? It adds additional complexity

Reply:
Yeah it's needed because 7zip library requires a Stream with the contents when you create a new file to add to the archive. So the usual API steps "create an empty file", "open file as readwrite", "write contents" can't be used.

Reply:
😫